packages/editor-ui/src/views/NodeView.vue

```diff
@@ -1871,6 +1875,53 @@ export default mixins(
 				this.deselectAllNodes();
 				this.nodeSelectedByName(newName);
 			},
+			checkCredentials(node: INodeUi) {
+				if (!node.credentials) {
+					return;
+				}
+				Object.keys(node.credentials).forEach((nodeCredentialType: string) => {
+					const credentialOptions = this.$store.getters['credentials/getCredentialsByType'](nodeCredentialType) as ICredentialsResponse[];
+					let nodeCredentials = node.credentials![nodeCredentialType];
+
+					// Check if workflows applies old credentials style
+					if (typeof nodeCredentials === 'string') {
+						nodeCredentials = {
+							id: null,
+							name: nodeCredentials,
+						};
+					}
+
+					if (nodeCredentials.id) {
+						// Check whether the id is matching with a credential
+						const credentialsForId = credentialOptions.find((optionData: ICredentialsResponse) => optionData.id === nodeCredentials.id);
+						if (credentialsForId) {
+							if (credentialsForId.name !== nodeCredentials.name) {
+								node.credentials![nodeCredentialType].name = credentialsForId.name;
+								this.credentialsUpdated = true;
+							}
+							return;
+						}
+					}
+
+					// No match for id found or old credentials type used
+					node.credentials![nodeCredentialType] = nodeCredentials;
+
+					// check if only one option with the name would exist
+					const credentialsForName = credentialOptions.filter((optionData: ICredentialsResponse) => optionData.name === nodeCredentials.name);
+
+					// only one option exists for the name, take it
+					if (credentialsForName.length === 1) {
+						node.credentials![nodeCredentialType].id = credentialsForName[0].id;
+						this.credentialsUpdated = true;
+						return;
+					}
+
+					// credentials have been following the old style, so update to new
+					if (nodeCredentials.id === null) {
```

Comment:
is not it possible to import a workflow with an id null?

Reply:
thx for pointing this out, this can be moved to the function where we check whether the string format is used 👍🏼  thanks!